source/manuals/programming-languages/html.html.md.erb

```diff
@@ -1,85 +1,223 @@
 ---
 title: HTML coding style
-last_reviewed_on: 2016-02-03
+last_reviewed_on: 2019-05-24
 review_in: 6 months
 ---
 
 # <%= current_page.data.title %>
 
-## HTML5
+## Progressive enhancement
 
-* Always quote attribute values - the decision tree for remember when to
-  selectively quote is too much mental overhead, so just quote them
-* Always use well-formed HTML - implicitly closed elements such as
-  `<p>` cause unnecessary parsing overhead and can cause issues in
-  accessibility
+HTML should be written in such a way that conforms with our [browser support model](https://github.com/alphagov/govuk-frontend#browser-support).
 
-## Use [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) roles appropriately
+For example, if IE8 is a supported browser, and it doesn't support the `<video>`
+tag, you should ensure an appropriate fallback is in place. Example:
 
-### Guidance on using [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) landmarks
+```html
+<video width="320" height="240" controls>
+  <source src="movie.mp4" type="video/mp4">
+  <source src="movie.ogg" type="video/ogg">
+  Your browser does not support the video tag.
+</video>
+```
 
-When you use HTML5 sectioning elements it's a good idea to include
-the appropriate [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) landmark roles as well. The trick when you're
-using both HTML5 and [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) is to put the landmark on the sectioning
-element itself because this avoids the information being duplicated by
-screen readers that support both technologies.
+When using structual HTML5 elements, such as `<main>`, ensure that you include
+a shiv that allows them to be styled in legacy browsers:
 
-Example:
+```html
+<!--[if lt IE 9]><script src="https://assets.publishing.service.gov.uk/static/ie-a4524544a53d57a7e259b4bb966b9c32557c98c920b77e52d09304642b68401a.js" crossorigin="anonymous" integrity="sha256-pFJFRKU9V6fiWbS7lmucMlV8mMkgt35S0JMEZCtoQBo="></script><![endif]-->
+```
 
-    <header role="banner"></header>
+## Element structure
 
-The following roles map directly to HTML5 sectioning elements:
+### Self-closing tags
 
-* `role="main"` maps to the `<main>` element
-* `role="complementary"` maps to the `<aside>` element
-* `role="contentinfo"` maps to the `<footer>` element
-* `role="navigation"` maps to the `<nav>` element
+Always use well-formed HTML. Whilst [end slashes are optional in HTML5](https://www.w3.org/TR/html5/syntax.html#start-tags),
+some tags _must_ be closed, so for consistency and easier parsing, it's safer
+to always close your tags. Where it is possible to self-close a tag, use self
+closing, otherwise use normal closing.
```

Comment:
Not sure about this advice.

`<img>`, `<br>` and friends are void elements: they don’t have closing tags. I’m not sure the advice to self-close these tags is useful. You still have to know it’s a void element becauuse you can’t do `<img src="foo.png" alt="Foo"></img>`. Similarly, you can’t have a self-closing non-void element like `<p/>` (or `<script src="foo.js"/>`, which is the one that catches lots of people out). 

The previous version of this guidance talked about [implicitly-closed elements](https://www.w3.org/TR/html5/syntax.html#optional-start-and-end-tags) like the `<dt>` and `<dd>` elements in `<dl><dt>kitten<dd>baby cat<dt>puppy<dd>baby dog</dl>`. The guidance to always close these elements with end tags is fairly useful because it reduced mental overhead.

(Also, this bit conflates tags and elements somewhat.)

Reply:
Thanks Alex, that's useful feedback 👍 I'll try and get my use of "tag" and "element" consistent.

You say you have to know whether or not an element is a void element, but that's only true of when we're _writing_ HTML. When we're reading it, we should be able to infer whether or not it's a void element through the presence of closing slashes.

For example, is `<made-up-element>` a void element or does its closing tag appear later in the document? It's clearer to do `<made-up-element />` if it's a void element, so that if we _do_ encounter `<another-made-up-element>` and there is no trailing slash, then we know it's a non-void element.

I think we're kind of saying the same thing, but I need to go back and write that portion of the guide more clearly.

Reply:
Have rewritten this section. I still advocate an explicit closing `/>`, hopefully the explanation is now clearer and the use of tag vs element now correct.